force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,46 @@
+trigger AccountTrigger on Account (before insert, after insert) {
+        // Before Insert Logic
+    if(Trigger.isBefore && Trigger.isInsert){
+        for(Account acc : Trigger.new){
+        // Set the Account Type to 'Prospect' if it's null
+        if(acc.Type == null){
+            acc.Type = 'Prospect';
+        }
+        
+ 	    // Copy Shipping Address to Billing Address if Shipping fields are not empty
+ 	    if(acc.ShippingStreet != null || 
+           acc.ShippingCity != null ||
+           acc.ShippingState != null ||
+           acc.ShippingPostalCode != null ||
+           acc.ShippingCountry != null){
+                acc.BillingStreet = acc.ShippingStreet;
+                acc.BillingCity = acc.ShippingCity;
+                acc.BillingState = acc.ShippingState;
+                acc.BillingPostalCode = acc.ShippingPostalCode;
+                acc.BillingCountry = acc.ShippingCountry;
+        }
```

Comment:
Because of the OR that you are using, it's possible for one of these fields to be Null and the code will still run and update.
I know it looks ugly but this is the best way to do this
if (acc.ShippingStreet != null) {
				acc.BillingStreet = acc.ShippingStreet;
			}
![Screenshot 2024-12-24 202546](https://github.com/user-attachments/assets/fef79908-f199-4902-a7a1-7851408f666d)